### DISCUSSION TITLE
Where to set image W and H when converting PDF to Image, PdfReadDefines.FitPage VS MagickImage.Resize

### DISCUSSION THREAD

**Discussion by tomasr78** [dlemstra/Magick.NET#860] [Help]
I am converting PDF to Image and a little confused what is the proper way to set the final Image dimension.

I have tried to do that using `PdfReadDefines `class and assign `MagickGeometry` to `PdfReadDefines.FitPage` property, it reduces the image dimension but not to 400px.  

```
var pdfReadDefines = new PdfReadDefines();
var magickGeometry = new MagickGeometry(400, 400) //Resize to max 400 one of dimensions and keep image proporition
       {
           IgnoreAspectRatio = false,
            Greater = true
         };
pdfReadDefines.Interpolate = true;
pdfReadDefines.FitPage = magickGeometry;

var magickReadSettings = new MagickReadSettings
{
   Density = new Density(300,300),
   ColorSpace = ColorSpace.sRGB
};

magickReadSettings.Defines = pdfReadDefines;
```

To solve this, I am resizing the image one more time using the method `MagickImage.Resize(magickGeometry)`.

I would like to understand what `pdfReadDefines.FitPage` does and I know that GhostScript accepts image dimension properties, then how to set it?

**Comment by dlemstra** [comment]
When you specify the fitpage the image will be fit inside the dimensions that you are specifying. The maximum size will be 400x400  in your example.

**Reply by tomasr78** [reply]
I have found that MagickReadSettings.Density property changes the geometry of an image. If I leave Density not set the image has correct geometry like it is set in MagickGeometry class, but if I set Density the geometry changes.

**Reply by dlemstra** [reply]
That is happening because setting the width and height internally changes the density (when its not set)
